src/NosCore.GameObject/Providers/QuestProvider/QuestProvider.cs

```diff
@@ -123,38 +137,75 @@ private async Task<bool> IsValidScriptAsync(ICharacterEntity character, QuestAct
             {
                 return false;
             }
+
             return script.StepType switch
             {
-                "q_complete" => ValidateQuest(character, script.Argument1 ?? 0),
-                "quest" => (character.Script?.ScriptId != scriptId
-                    || character.Script?.ScriptStepId != scriptStepId
-                    || type != QuestActionType.Validate
-                    || !script.Argument1.HasValue
-                    || await AddQuestAsync(character, (short)script.Argument1).ConfigureAwait(false)
-                ),
+                "q_complete" => await ValidateQuestAsync(character, script.Argument1 ?? 0).ConfigureAwait(false),
+                "quest" => await AddQuestAsync(character, type, script.Argument1 ?? 0).ConfigureAwait(false),
+                "time" => await TimeAsync(script.Argument1 ?? 0).ConfigureAwait(false),
+                "targetoff" => await TargetOffPacketAsync(script.Argument1 ?? 0, character).ConfigureAwait(false),
                 "web" => true,
                 "talk" => true,
-                "target" => true,
                 "openwin" => true,
-                "opendual" => false,
-                "time" => false,
-                "move" => false,
-                "run" => true,
-                "q_pay" => true,
-                "targetoff" => true,
+                "opendual" => true,
+
+                "move" => false, //todo handle
+                "q_pay" => false, //todo handle
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/manual/NosCoreIO/NosCore/pullRequest?prid=5304413)